rcm.html

```diff
@@ -1,118 +1,90 @@
+<!DOCTYPE html>
 <html>
-<div id="result" style="font-size: large"></div>
-<script>
-
+  <body>
+    <div id="result" style="font-size: large"></div>
+    <script type="application/javascript">
 // this program does a lot of random selection of array elements.
-function get_random_element(x)
-{
-  return x[Math.floor(Math.random() * x.length)];
-}
+const getRandomElement = (x) => x[Math.floor(Math.random() * x.length)];
 
-function generate_new_practice()
-{
+const generateNewPractice = () => {
   // lhs is the full enumeration of the scales I need to memorize
   // rhs is its half-step "pair"
   // I have found it helpful to study them as pairs
-  var scales = [];
-  scales["G Major"]      = "A-flat Major";
-  scales["A-flat Major"] = "G Major";
-  scales["D Major"]      = "D-flat Major";
-  scales["D-flat Major"] = "D Major";
-  scales["E Minor"]      = "F Minor";
-  scales["F Minor"]      = "E Minor";
-  scales["B Minor"]      = "B-flat Minor";
-  scales["B-flat Minor"] = "B Minor";
-
-  var scale1 = get_random_element(Object.keys(scales));
-  var scale2 = scales[scale1];
+  const scales = {
+    'G Major'      : 'A-flat Major',
+    'A-flat Major' : 'G Major',
+    'D Major'      : 'D-flat Major',
+    'D-flat Major' : 'D Major',
+    'E Minor'      : 'F Minor',
+    'F Minor'      : 'E Minor',
+    'B Minor'      : 'B-flat Minor',
+    'B-flat Minor' : 'B Minor',
+  };
 
+  const scale1 = getRandomElement(Object.keys(scales));
+  const scale2 = scales[scale1];
   // the exam requires a lot of variations of the scales
-  tremolando = Math.random() > (2/3); // equal practice for triplets, sixteenth notes and tremolando  
-  if (tremolando)
-  {
-	var note_value = "triplet @ 88";
-  }
-  else
-  {
-	var note_value = get_random_element(["triplet @ 69", "16th note @ 69"]);
-  }
-  var stroke     = get_random_element(["Rest Stroke", "Free Stroke"]);
-  var fingering  = get_random_element(["im", "ma", "ia"]);
+  const tremolando = Math.random() > (2 / 3); // equal practice for triplets, sixteenth notes and tremolando
 
+  const noteValue = tremolando ? 'triplet @ 88' : getRandomElement(['triplet @ 69', '16th note @ 69']);
```

Comment:
the `const` means that variables can only be assigned once, reducing potential side effects and bugs, so you tend to use ternaries more